system/Validation/Rules.php

```diff
@@ -436,6 +436,47 @@ public function required_without(
         return true;
     }
 
+    /**
+     * This field is required when any of the other required fields have their expected values present
+     * in the data.
+     *
+     * Example (The special_option field is required when the normal_option,1,2 field has the given value.):
+     *
+     *     required_if[normal_option,1,2]
+     *
+     * @param string|null          $str
+     * @param string|null          $fieldWithValue that we should check if present
+     * @param array<string, mixed> $data           Complete list of field from the form
+     */
+    public function required_if($str = null, ?string $fieldWithValue = null, array $data = []): bool
+    {
+        if ($fieldWithValue === null || $data === []) {
+            throw new InvalidArgumentException('You must supply the parameters: field,expected_values, data.');
```

Comment:
The message is not reflecting the correct arguments.

Reply:
How about this?
```php
throw new InvalidArgumentException("You must supply the field value: {$field},expected_values?");
```

Reply:
How about this?
```php
throw new InvalidArgumentException('You must supply the parameters: field,value1,value2,...');
```

Reply:
IMO better.